docs/proposal-rose-suite-run.md

```diff
@@ -136,16 +236,28 @@ sources on installation. Other things to consider:
 The `rose suite-run` command calls `cylc validate --strict` by default.
 Automatic suite validation should become the default behaviour for `cylc run`
 and `cylc reload`. Things to consider:
-* Is the *strict* mode necessary by default?
-* `rose suite-run` does not currently validate Rose items. This should be
-  changed to ensure that all aspects of the suite are validated uniformly.
+* `rose suite-run` does not currently validate Rose apps against their
+  metadata. Rose apps should be validated against their metadata by default,
```

Comment:
To clarify are you suggesting a Rose plugin to Cylc which validates all applications during the suite install process?

Conceptually sounds good to me but we may need to consult users before considering this as it is potentially controversial.

Rose has different levels of validation:

* Built-in (fast)
* Fail-if Warn-if (fastish)
* Macros (slow)

What level would we validate to? Would we validate optional configs?

We might want to upgrade `rose macro` to use asyncio or multiprocessing.

Reply:
I believe that is the general plan, but I agree, there is the possibility of some controversy. These are all good points, and I have amended the document to reference them.